pkg/storage/etcd3/watcher.go

```diff
@@ -180,6 +180,9 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	wc.cancel()
```

Comment:
Could you back this all into stop to make it explicit to other folks.   Also where does the channel get closed on the fall-through?